security/fma-contract-version-upgrade.md

```diff
@@ -0,0 +1,122 @@
+
+    According to the solidity blog, this bug can only occur if a contract contains pattern:
+
+    1. A storage write. Note that the write may still be removed even if it happens only conditionally or within a call to a function that ends up being inlined.
+    2. A call to a function that conditionally terminates using inline assembly as described above, but also has a different code path that returns to the caller.
+    3. Any continuing control flow path does one of the following:
+       - It overwrites the storage write in (1).
+       - It reverts.
+
+- **Risk Assessment:**
+  - Severity: High (if present)
+  - Likelihood: Low
+- **Mitigations:** Use a solidity version above 0.8.17.
+- **Detection:** If a solidity contract (using solidity version earlier than 0.8.17) follows the pattern described above, its potentially vulnerable to this.
+- **Recovery Path(s)**: Redeployment and upgrade of the affected contracts
+
+### Change in Default EVM version between different Solidity versions
+
+- **Description:** Version 0.8.25 sets the default evm version to cancun
+- **Risk Assessment:**
+  - Severity: Low/Medium/High
+  - Likelihood: Medium
+- **Mitigations:** As at 0.8.15, the default EVM version is 'London', if no explicit version override is declared, 'Cancun' will be used rather which if not intended can lead to code that either does not deploy or reverts un-ideally at runtime even though non-fork tests might have passed. A mitigation is explicitly declaring the intended EVM version to use for compilation regardless of if its the same as the default the compiler uses. That way, a new compiler version that changes the default EVM version does not change the one that is intended to be used. In this scenario however, this is not possible because the intended EVM version is the same as the compiler's EVM version default.
+- **Detection:** Running unit tests as fork test also.
+- **Recovery Path(s)**: Redeployment and upgrade of the affected contracts
+
+## Audit Requirements
+
+_Will this project require an audit according to the guidance in [OP Labs Audit Framework: When to get external security review and how to prepare for it](https://gov.optimism.io/t/op-labs-audit-framework-when-to-get-external-security-review-and-how-to-prepare-for-it/6864)? Please explain your reasoning._
+
+## Action Items
+
+Below is what needs to be done before launch to reduce the chances of the above failure modes occurring, and to ensure they can be detected and recovered from:
+
+- [ ] Resolve all comments on this document and incorporate them into the document itself (Assignee: document author)
```

Comment:
Any other action items we add here? Two that come to mind are testing the stack too deep and code size situations from whatever new version we choose (ref https://github.com/ethereum-optimism/design-docs/pull/69/files?diff=unified&w=0#r1746167621)

Reply:
New actions LGTM, one more action could be to decide on a solidity version, based on the new FMA item